src/core/listener.c

```diff
@@ -406,6 +406,7 @@ QuicListenerIndicateEvent(
     _Inout_ QUIC_LISTENER_EVENT* Event
     )
 {
+    CXPLAT_PASSIVE_CODE();
```

Comment:
If you want to save a few bytes of memory, you could make each of these purely `CXPLAT_PASSIVE_CODE` functions pageable, so the linker puts them into a different code segment.

Reply:
What exactly does that get us? What does it cost? Does it mean the function can never raise the irql?

Reply:
Right, if the function ever executes at dispatch or higher, it cannot be pageable. You can also lock and unlock paged sections at runtime, e.g. if you go from zero to one connections.

Reply:
Marking the function pageable won't reliably detect irql violations right? It would AV only when the pages are paged out?

Reply:
Right, driver verifier will occasionally page out pageable pages, but otherwise you'd have to rely on PAGED_CODE or PASSIVE_CODE asserts to detect IRQL violations. The benefit of making it pageable is primarily to be friendly to the memory manager.